integration-tests/src/tests/client/state_snapshot.rs

```diff
@@ -127,6 +126,40 @@ fn test_maybe_open_state_snapshot_garbage_snapshot() {
     assert!(result.is_err());
 }
 
+#[test]
+fn test_state_snapshot_disabled() -> anyhow::Result<()> {
+    init_test_logger();
+    let genesis = Genesis::test(vec!["test0".parse().unwrap()], 1);
+    let env = TestEnv::builder(&genesis.config)
+        .clients_count(1)
+        .real_stores()
+        .nightshade_runtimes(&genesis)
+        .build();
+
+    let genesis_block = env.clients[0].chain.get_block_by_height(0)?;
+
+    let store = env.clients[0].chain.chain_store().store();
+    let state_snapshot_test_env =
+        set_up_test_env_for_state_snapshots(&store, StateSnapshotType::Disabled);
+
+    if std::fs::exists(&state_snapshot_test_env.state_snapshots_dir)? {
+        std::fs::remove_dir_all(&state_snapshot_test_env.state_snapshots_dir)?;
+    }
+
+    state_snapshot_test_env.shard_tries.create_state_snapshot(
+        CryptoHash::default(),
+        &[(0, ShardUId::single_shard())],
+        &genesis_block,
+    )?;
+
+    anyhow::ensure!(
```

Comment:
I'm not familiar with this, how is it better than simple assertion?

Reply:
It returns `anyhow::Error` instead of panicking. As the function already uses `anyhow::Result`, I though it would be an appropriate way to assert.